geth/jail/handlers.go

```diff
@@ -66,23 +66,54 @@ func registerHandlers(jail *Jail, cell common.JailCell, chatID string) error {
 	return nil
 }
 
+// makeAsyncSendHandler returns jeth.sendAsync() handler.
+func makeAsyncSendHandler(jail *Jail, cellInt common.JailCell) func(call otto.FunctionCall) otto.Value {
+	// FIXME(tiabc): Get rid of this.
+	cell := cellInt.(*Cell)
+	return func(call otto.FunctionCall) otto.Value {
+		go func() {
+			response := jail.Send(call, cell.VM)
```

Comment:
Doesn't the same comments as on line :92 (in `makeSendHandler` - `Send calls are guaranteed to be only invoked...`) applies here?  Should not it unlock and lock mutex as well here?

Reply:
Nope. Moreover, we'll get panic if we do this.
The reason is that we run this in a goroutine and just return `otto.UndefinedValue()` and this is fast. The cell will be quickly unlocked after these 2 operations and `jail.Send` execution in the goroutine will start afterwards.

Reply:
Ok. I'm not sure I completely understand workflow here, but fine for now.

Reply:
This is an important piece, let's sort it out. I may also turn out to be wrong here and it will be fatal. I'll draw a diagram in a minute.

Reply:
![send and sendasync](https://user-images.githubusercontent.com/2167748/30524112-1ca3782e-9bf6-11e7-803d-75084a7d2fe6.png)

Reply:
Something like that:
![send and sendasync](https://user-images.githubusercontent.com/2167748/30524147-d0f8ff56-9bf6-11e7-8337-02f269326b35.png)

Reply:
Woah, this is a cool picture explaining a lot. Let's add it do the documentation doc. 

Just one question with calls without callbacks: if a long running operation will be called and another independent call without callback will be executed, we are still fine because we use thread-safe otto (our `vm.VM` struct)?

Reply:
Yes, these two will be just run in sequence.